source/beamline/AMBeamline.cpp

```diff
@@ -192,3 +194,27 @@ void AMBeamline::initializeBeamlineSupport(){
 	AMBeamlineSupport::setBeamlineDetectorSetAPI(AMBeamline::bl());
 	AMBeamlineSupport::setBeamlineSynchronizedDwellTimeAPI(AMBeamline::bl());
 }
+
+AMAction3* AMBeamline::createScanInitializationAction(AMGenericStepScanConfiguration *configuration)
```

Comment:
We might need to specify the function name here. Basically, we are creating the scan initialization actions for a AMGenericStepScan. Therefore, it might be better to name the function `createGenericStepScanInitializationAction()`.

At this point of time, only BioXAS scan configurations are inherited from the AMGenericStepScanConfiguration. All the other beamlines are inherited from AMStepScanConfiguration. Therefore, I don't think it is a good idea to put this function in the AMBeamline. It might be ideal to live in the corresponding scanConfiguraitonControl.